### DISCUSSION TITLE
Combining `DisplayFromStr` and `Uppercase`

### DISCUSSION THREAD

**Discussion by ctron** [jonasbb/serde_with#419] [Q&A] [answered]
I am trying to de-serialize into a type using `from_str`, however I would like the string to upper all upper cases before handing over the data to `from_str`.

I noticed there is a "format" `Uppercase`, but I couldn't find any example on how to use it.

**Comment by jonasbb** [comment, marked as answer]
Hi, that is currently not possible. The `DisplayFromStr` does not allow any modification of the serialized string. I don't think it necessarily makes sense to add support for that in this crate. It is simple enough to create your own custom `MyDisplayFromStr` type with any string modification operation you need, instead of a limited pre-defined set. You could also define a generic type like `MyDisplayFromStr<Uppercase>` if you need that flexibility. Uppercasing a String is also so complicated that it has different meanings, so picking one probably doesn't work too well.

`Uppercase` cannot be combined with any arbitrary other type. The types in the `format` module are only needed if some other type needs them. For example, you can specify the formatting of your hex values https://docs.rs/serde_with/latest/serde_with/hex/struct.Hex.html. `Hex` doesn't have the same String uppercasing problems, since we have a limited characterset here.
